lib/IRGen/Fulfillment.cpp

```diff
@@ -124,6 +124,15 @@ bool FulfillmentMap::searchTypeMetadata(IRGenModule &IGM, CanType type,
                                            source, MetadataPath(path), keys);
     }
 
+    // If the type is an archetype consider its super class bound.
+    if (isa<ArchetypeType>(type))
+      if (metadataState == MetadataState::Complete) {
```

Comment:
Maybe this should be the same condition as below, i.e. `metadataState == MetadataState::Complete && keys.isInterestingType(type))` rather than checking specifically for `ArchetypeType`.

Reply:
I will lift the restriction. Is interestingType(type) is checked earlier.